### PR TITLE
samples(recaptcha_enterprise): add samples for site key

### PR DESCRIPTION
**Pull request by NivedhaSenthil**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-cloud-ruby/issues) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea.
- [x] Follow the instructions in [CONTRIBUTING](CONTRIBUTING.md). Most importantly, ensure the tests and linter pass by running `bundle exec rake ci` in the gem subdirectory.
- [x] Update code documentation if necessary.

### CODE REVIEW

File: google-cloud-recaptcha_enterprise/samples/acceptance/site_key_test.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    @site_keys.each do |site_key|
+      client.delete_key name: site_key
+    end
+  end
+
+  def create_test_site_key
+    create_key_request = { parent: "projects/#{project_id}",
+        key:
+         { display_name: "test_key",
+          web_settings: { integration_type: 1, allowed_domains: ["localhost"], allow_amp_traffic: false } } }
```
Comment (dazuma) [inline google-cloud-recaptcha_enterprise/samples/acceptance/site_key_test.rb:45]: Should be able to make this more readable.

File: google-cloud-recaptcha_enterprise/samples/recaptcha_enterprise_create_site_key.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  request = { parent: "projects/#{project_id}",
+      key: {
+        display_name: "any descriptive name for the key",
+        web_settings: {
+          allowed_domains: [domain],
+            allow_amp_traffic: false,
+            integration_type: 1
+        }
+      } }
```
Comment (dazuma) [inline google-cloud-recaptcha_enterprise/samples/recaptcha_enterprise_create_site_key.rb:35]: Can you clean up the indentation etc a bit to make this more readable? Maybe like:

```ruby
  request = {
    parent: "projects/#{project_id}",
    key: {
      display_name: "any descriptive name for the key",
      web_settings: {
        allowed_domains: [domain],
        allow_amp_traffic: false,
        integration_type: 1
      }
    }
  }
```

File: google-cloud-recaptcha_enterprise/samples/recaptcha_enterprise_update_site_key.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  # Create the reCAPTCHA client.
+  client = ::Google::Cloud::RecaptchaEnterprise.recaptcha_enterprise_service
+
+  request_key = client.get_key name: "projects/#{project_id}/keys/#{site_key}"
+  request_key.web_settings.allowed_domains.push domain
+  request_key.web_settings.allow_amp_traffic = true
+
+  client.update_key key: request_key
+
+  sleep 2
```
Comment (dazuma) [inline google-cloud-recaptcha_enterprise/samples/recaptcha_enterprise_update_site_key.rb:34]: Add a comment explaining the sleep. (Maybe something like "Key updates can take a few seconds to propagate.")

File: google-cloud-recaptcha_enterprise/samples/recaptcha_enterprise_update_site_key.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+  # Retrieve the key and check if the property is updated.
+  response = client.get_key name: "projects/#{project_id}/keys/#{site_key}"
+  web_settings = response.web_settings
+
+  # Get the changed property.
+  if web_settings.allow_amp_traffic
+    puts "reCAPTCHA Site key successfully updated !"
+  else
+    puts "Error! reCAPTCHA Site key property hasn't been updated. Please try again !"
```
Comment (dazuma) [inline google-cloud-recaptcha_enterprise/samples/recaptcha_enterprise_update_site_key.rb:44]: When would this actually happen? Are we telling the user this is an expected outcome? How should they respond if it does happen?

**Review by dazuma — CHANGES_REQUESTED** [review]

File: google-cloud-recaptcha_enterprise/samples/recaptcha_enterprise_update_site_key.rb (hunk 4 above)
Comment (NivedhaSenthil) [reply google-cloud-recaptcha_enterprise/samples/recaptcha_enterprise_update_site_key.rb]: Added these just to be similar with other language samples and doesn't really affect the functionality hence removing them.

**Review by shivgautam — APPROVED** [review]

**Review by dazuma — APPROVED** [review]

**Pull request merged**